lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -2380,9 +2382,15 @@ class PrimalGenCloner final
 
   void visitApplyInst(ApplyInst *ai) {
     auto &context = getContext();
-    // Special handling logic only applies when `apply` is active. If not, just
-    // do standard cloning.
-    if (!activityInfo.isActive(ai, synthesis.indices)) {
+    // Special handling logic only applies when `apply` has active results. If
+    // not, just do standard cloning.
+    SmallVector<SILValue, 4> allResults;
+    allResults.push_back(ai);
```

Comment:
`ai` could be a tuple of multiple results. Use `extractElements` first.

Reply:
I initially tried `extractElements`, but I changed it to this when I found `extractElements` was problematic. I don't believe activity analysis destructures tuple results, it treats them as one value.

Reply:
Ah, sorry i didn't mean `extractElements`, but `collectAllActualResultsInTypeOrder`. `extractElements` would definitely not work because it generates new instructions which will not have any activity information (they wouldn't even be in the original function).

Reply:
I see. I think `collectAllActualResultsInTypeOrder` expects an `ArrayRef<SILValue>` of tuple-extracted direct results though, based on its current caller and some print debugging.

```cpp
    // Extract all results from `pullbackCall`.
    SmallVector<SILValue, 8> dirResults;
    extractAllElements(pullbackCall, builder, dirResults); // <--- call to `extractAllElements`
    // Get all results in type-defined order.
    SmallVector<SILValue, 8> allResults;
    collectAllActualResultsInTypeOrder(
        pullbackCall, dirResults, pullbackCall->getIndirectSILResults(),
        allResults);
```

So I think what we want is to do sth like `collectAllActualResultsInTypeOrder` (to collect indirect results), except that if any `SILResultInfo` is direct, then we append `ai` (the entire direct result) itself?

Reply:
Yes!